codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/XmlBindingTraitParserGenerator.kt

```diff
@@ -201,7 +200,7 @@ class XmlBindingTraitParserGenerator(
     }
 
     override fun errorParser(errorShape: StructureShape): RuntimeType {
-        val fnName = errorShape.id.name.toString().toSnakeCase()
```

Comment:
I _think_ this is a bug—it will duplicate parsers between an error shape and the same structure used inside of a shape.

Reply:
(the code before added `_inner` to the structure parser which disambiguated)

Reply:
I think you're right. It would be incredibly uncommon, where a model is using the exact same struct shape as both an error and non-error, but a bug none the less.